cmd/bindings-forwarder/main.go

```diff
@@ -0,0 +1,161 @@
+	"github.com/ngrok/ngrok-operator/internal/version"
+	"github.com/ngrok/ngrok-operator/pkg/bindingsdriver"
+	//+kubebuilder:scaffold:imports
+)
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(bindingsv1alpha1.AddToScheme(scheme))
+}
+
+func main() {
+	if err := cmd().Execute(); err != nil {
+		setupLog.Error(err, "error running bindings-forwarder-manager")
+		os.Exit(1)
+	}
+}
+
+type managerOpts struct {
+	// flags
+	metricsAddr string
+	probeAddr   string
+	description string
+	managerName string
+	zapOpts     *zap.Options
+
+	// env vars
+	namespace string
+}
+
+func cmd() *cobra.Command {
+	var opts managerOpts
+	c := &cobra.Command{
+		Use: "bindings-forwarder-manager",
+		RunE: func(c *cobra.Command, args []string) error {
+			return runController(c.Context(), opts)
+		},
+	}
+
+	c.Flags().StringVar(&opts.metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to")
+	c.Flags().StringVar(&opts.probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	c.Flags().StringVar(&opts.description, "description", "Created by the ngrok-operator", "Description for this installation")
+	c.Flags().StringVar(&opts.managerName, "manager-name", "bindings-forwarder-manager", "Manager name to identify unique ngrok operator agent instances")
+
+	opts.zapOpts = &zap.Options{}
+	goFlagSet := flag.NewFlagSet("manager", flag.ContinueOnError)
+	opts.zapOpts.BindFlags(goFlagSet)
+	c.Flags().AddGoFlagSet(goFlagSet)
+
+	return c
+}
+
+func runController(_ context.Context, opts managerOpts) error {
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(opts.zapOpts)))
+
+	buildInfo := version.Get()
+	setupLog.Info("starting bindings-forwarder-manager", "version", buildInfo.Version, "commit", buildInfo.GitCommit)
+
+	var ok bool
+	opts.namespace, ok = os.LookupEnv("POD_NAMESPACE")
+	if !ok {
+		return errors.New("POD_NAMESPACE environment variable should be set, but was not")
+	}
+
+	options := ctrl.Options{
+		Scheme: scheme,
+		Cache: cache.Options{
+			DefaultNamespaces: map[string]cache.Config{
+				opts.namespace: {},
+			},
+		},
+		Metrics: server.Options{
+			BindAddress: opts.metricsAddr,
+		},
+		WebhookServer:          webhook.NewServer(webhook.Options{Port: 9443}),
+		HealthProbeBindAddress: opts.probeAddr,
+		LeaderElection:         false,
+	}
+
+	// create default config and clientset for use outside the mgr.Start() blocking loop
+	k8sConfig := ctrl.GetConfigOrDie()
+	mgr, err := ctrl.NewManager(k8sConfig, options)
+	if err != nil {
+		return fmt.Errorf("unable to start bindings-forwarder-manager: %w", err)
+	}
+
+	bd := bindingsdriver.New()
+
+	if err = (&bindingscontroller.ForwarderReconciler{
+		Client:         mgr.GetClient(),
+		Log:            ctrl.Log.WithName("controllers").WithName("tunnel"),
```

Comment:
Nit: Is this a tunnel controller or some other kind?

Reply:
Looks like a copypasta mistake. I'll update it